servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/dataFetchingEnvironmentExtensions.kt

```diff
@@ -29,17 +29,25 @@ import java.util.concurrent.CompletableFuture
  * Helper method to get a value from a registered DataLoader.
  * The provided key should be the cache key object used to save the value for that particular data loader.
  */
+@Deprecated(
+    message = "Custom DataLoader extension functions are deprecated and will be removed in next major release",
+    replaceWith = ReplaceWith("getDataLoader<K, V>(dataLoaderName)")
+)
 fun <K, V> DataFetchingEnvironment.getValueFromDataLoader(dataLoaderName: String, key: K): CompletableFuture<V> {
     val loader = getDataLoader<K, V>(dataLoaderName) ?: throw MissingDataLoaderException(dataLoaderName)
-    return loader.load(key, this.getContext())
+    return loader.load(key, this.graphQlContext)
 }
 
 /**
 * Helper method to get values from a registered DataLoader.
 */
+@Deprecated(
```

Comment:
These methods are useful. I don't think we need to deprecate these. I was referring to the methods to get values from context directly on `DataFetchingEnvironment`, ie lines 57+ in this file

Reply:
ahh ok, well we added those extensions bc of a feature request, where ppl stores in context using KClass, so is definetely more idiomatic do to

```kotlin
dataFetchingEnvironment.getFromContext<CoroutineScope>()
```

than

```kotlin
dataFetchingEnvironment.graphQLContext.get(CoroutineScope::class)
```

Reply:
But with the extension methods [here](https://github.com/ExpediaGroup/graphql-kotlin/blob/415a745254262c8757b0805c2e6a85e6fba9a485/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/extensions/GraphQLContextExtensions.kt#L26-L27) you can do this

```kotlin
dataFetchingEnvironment.graphQLContext.get<CoroutineScope>()
```

Reply:
i still find it useful, but yes, it looks like an unneeded indirection, i can remove these in a separate PR.

Reply:
I don't think they should be deleted just yet, maybe just deprecated, but yes that can be a separate PR